### PR TITLE
[11.0][MIG] analytic_tag_dimension

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Maybe we should add `account_accountant` as a dependency? Looks like there is no sense in this module until `account_accountant` is installed and so user is able to turn analytic accounting on in invoicing settings without abusing tech features.

**Comment by pedrobaeza** [comment]
That module doesn't exist in community. You have to enable a permission in the user.

**Comment by simahawk** [comment]
@pedrobaeza hola (and happy new year ;)). One thing to mention: the `Invoicing` main menu is hidden in v11. Shouldn't we have a "stupid" micro module that enables it in community? And depend on it?

File: analytic_tag_dimension/README.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+To configure this module, you need to:
+
+* go to /Accounting/Configuration/Analytic Accounting/Analytic Accounts Dimensions to create new Analytic Dimensions.
+* You can create new Analytic Tags on this form, or go to /Accounting/Configuration/Analytic Accounting/Analytic Accounts Tags and set Dimension for each Tag.
+
+
+Known issues / Roadmap
+======================
+
+* Analytic Entries with Tags created before installing this module are not updated with its Dimensions.
```
Comment (simahawk) [inline analytic_tag_dimension/README.rst]: `s/with its/with theirs`

File: analytic_tag_dimension/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2017 PESOL (http://pesol.es)
+#                Angel Moya (angel.moya@pesol.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Analytic Accounts Dimensions",
+    'summary': "Group Analytic Entries by Dimensions",
+    "version": "11.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "PESOL, Odoo Community Association (OCA)",
+    "website": "http://pesol.es",
```
Comment (simahawk) [inline analytic_tag_dimension/__manifest__.py]: https://github.com/OCA/account-analytic

File: analytic_tag_dimension/demo/analytic_demo.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+  <record id="analytic_dimension_type" model="account.analytic.dimension">
+      <field name='name'>Type</field>
```
Comment (simahawk) [inline analytic_tag_dimension/demo/analytic_demo.xml]: mixed up quotes and double quotes. Keep the latter

**Comment by pedrobaeza** [comment]
Happy new year to you too!

Not exactly, the invoicing menu is shown when installing sale_managing if I don't remember bad. What is hidden until you enable the permission "Show full accounting features" is "Journal entries" menu for example. If you consider that the menu should be shown independently, go ahead with the module.

File: analytic_tag_dimension/models/analytic.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+class AnalyticDimensionLine(models.AbstractModel):
+    _name = 'analytic.dimension.line'
+    _analytic_tag_field_name = 'analytic_tag_ids'
+
+    @api.model
+    def create(self, values):
+        result = super(AnalyticDimensionLine, self).create(values)
+        if values.get(self._analytic_tag_field_name):
+            tag_ids = getattr(
```
Comment (simahawk) [inline analytic_tag_dimension/models/analytic.py]: don't use `getattr` -> `result[self._analytic_tag_field_name]`

File: analytic_tag_dimension/models/analytic.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                result, self._analytic_tag_field_name)
+            tag_ids._check_analytic_dimension()
+            dimension_values = tag_ids.get_dimension_values()
+            super(AnalyticDimensionLine, result).write(dimension_values)
+        return result
+
+    @api.multi
+    def write(self, values):
+        result = super(AnalyticDimensionLine, self).write(values)
+        if values.get(self._analytic_tag_field_name):
```
Comment (simahawk) [inline analytic_tag_dimension/models/analytic.py:90]: this piece of code is duplicated within create and write. I'd split it to a `_handle_analytic_dimension` multi method and call it from both create and write.

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by melroy89** [comment]
Off-topic:
@simahawk  @pedrobaeza I had the same remark, regarding the accounting module. I think the administrator for sure, needs to have "Show full accounting features" enabled by default, during setup of this module / account in Odoo. You should give the admin all the rights & views that are possible within Odoo. That is my opinion, especially if new people are trying the software, they are missing out some great hidden features...

**Comment by simahawk** [comment]
@pedrobaeza thinking more about the accounting hidden stuff issue: if we add a module I think it should stay into OCB repo to avoid ppl having to work out inter-dependencies across OCA repos just for that.
What do you think?

**Comment by simahawk** [comment]
@arkostyuk tests failing

**Comment by gurneyalex** [comment]
regarding the availability of the menu, I think a note in the README file saying something like "to properly configure this module, you need to enable xxx in the xxxx" is the minimum to explain.

File: analytic_tag_dimension/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2017 PESOL (http://pesol.es)
+#                Angel Moya (angel.moya@pesol.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (ghost) [inline analytic_tag_dimension/__manifest__.py:3]: Add `# Copyright 2018 Camptocamp`

**Review by ghost — APPROVED** [review]
LGTM

**Review by mpanarin — APPROVED** [review]
LGTM

**Comment by fclementic2c** [comment]
Works pretty well. 
But I found a bug when you do a group by on BI views (reporting / management / Invoices)
I noticed the same issue with Odoo 10 -> please correct this as well 
screenshot-3315761-146-8f7ee6 runbot1 odoo-community org-2018-01-26-13-04-33-719 1







Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 930, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 922, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/models.py", line 1828, in read_group
    result = self._read_group_raw(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
  File "/.repo_requirements/odoo/odoo/models.py", line 1918, in _read_group_raw
    self._cr.execute(query, where_clause_params)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: column account_invoice_report.x_dimension_concept does not exist
LINE 3: ..."account_invoice_report__x_dimension_concept" ON ("account_i...
                                                             ^

**Review by yvaucher — APPROVED** [review]
LGTM

**Pull request merged**